tests/foreman/ui/test_host.py

```diff
@@ -3127,3 +3127,87 @@ def _cleanup():
+
+    :expectedresults:
+        1. Multiple hosts can be selected in All Hosts page
+        2. Bulk organization/location change operation succeeds
+        3. Hosts are successfully moved to new organization/location
+        4. Host properties reflect the new taxonomies
+
+    :CaseComponent: Hosts
+
+    :Team: Phoenix-subscriptions
+
+    :CaseImportance: High
+    """
+    # Create second organization and location for the test
+    second_org = module_target_sat.api.Organization().create()
+    second_location = module_target_sat.api.Location(organization=[second_org]).create()
+
+    # Register both content hosts to first organization and location only
+    host_names = []
+    for host in content_hosts[:2]:  # Use first 2 content hosts
+        # Register host to first organization and location
+        result = host.register(function_org, function_location, None, module_target_sat)
+        assert result.status == 0, f'Failed to register host: {result.stderr}'
+
+        host_names.append(host.hostname)
+
+    # Verify hosts are initially in the first org/location
+    for host_name in host_names:
+        host_entity = module_target_sat.api.Host().search(query={'search': f'name={host_name}'})[0]
+        assert host_entity.organization.id == function_org.id
+        assert host_entity.location.id == function_location.id
+
+    # Perform bulk organization/location change through UI
+    with module_target_sat.ui_session() as session:
+        # Select the first organization to see the hosts
+        session.organization.select(function_org.name)
+        session.location.select(function_location.name)
+
+        # Navigate to All Hosts page and perform bulk change
+        session.all_hosts.change_organization_location(
+            host_names=host_names,
+            new_organization=second_org.name,
+            new_location=second_location.name,
+            option="Fix on mismatch",
+        )
+
+        # Switch to second organization to verify the change
+        session.organization.select(second_org.name)
+        session.location.select(second_location.name)
+
+        # Verify hosts appear in the new organization/location
+        for host_name in host_names:
+            host_values = session.all_hosts.read_host(host_name)
+            assert host_values is not None, f"Host {host_name} not found in new organization"
+
+    # API verification - verify hosts are now in the second org/location
+    for host_name in host_names:
+        host_entity = module_target_sat.api.Host().search(query={'search': f'name={host_name}'})[0]
+        assert host_entity.organization.id == second_org.id, (
+            f"Host {host_name} not moved to second organization"
+        )
+        assert host_entity.location.id == second_location.id, (
+            f"Host {host_name} not moved to second location"
+        )
+
+    # Cleanup - Delete the test organizations (locations will be cleaned up automatically)
+    second_org.delete()
```

Comment:
This will not happen automatically if the test fails before this point, I would suggest using something like this 
    
```suggestion
    # Cleanup - Delete the test organizations (locations will be cleaned up automatically)
    @request.addfinalizer
    def cleanup():
	second_org.delete()
```